server/build/webpack/base.config.js

```diff
@@ -0,0 +1,318 @@
+import path, {sep} from 'path'
+import fs from 'fs'
+import webpack from 'webpack'
+
+import nodeExternals from 'webpack-node-externals'
+import UglifyJSPlugin from 'uglifyjs-webpack-plugin'
+import CaseSensitivePathPlugin from 'case-sensitive-paths-webpack-plugin'
+import WriteFilePlugin from 'write-file-webpack-plugin'
+import FriendlyErrorsWebpackPlugin from 'friendly-errors-webpack-plugin'
+import ExtractTextPlugin from 'extract-text-webpack-plugin'
+import {getPages} from './utils'
+import CombineAssetsPlugin from '../plugins/combine-assets-plugin'
+import PagesPlugin from '../plugins/pages-plugin'
+import DynamicChunksPlugin from '../plugins/dynamic-chunks-plugin'
+import findBabelConfig from '../babel/find-config'
+
+const nextDir = path.join(__dirname, '..', '..', '..', '..')
+const nextNodeModulesDir = path.join(nextDir, 'node_modules')
+const nextPagesDir = path.join(nextDir, 'pages')
+const defaultPages = [
+  '_error.js',
+  '_document.js'
+]
+const interpolateNames = new Map(defaultPages.map((p) => {
+  return [path.join(nextPagesDir, p), `dist/bundles/pages/${p}`]
+}))
+
+function babelConfig (dir, isServer) {
+  const mainBabelOptions = {
+    cacheDirectory: Boolean(process.env.BABEL_DISABLE_CACHE),
+    presets: [],
+    plugins: [
+      !isServer && require.resolve('react-hot-loader/babel')
+    ].filter(Boolean)
+  }
+
+  const externalBabelConfig = findBabelConfig(dir)
+  if (externalBabelConfig) {
+    console.log(`> Using external babel configuration`)
+    console.log(`> Location: "${externalBabelConfig.loc}"`)
+    // It's possible to turn off babelrc support via babelrc itself.
+    // In that case, we should add our default preset.
+    // That's why we need to do this.
+    const { options } = externalBabelConfig
+    mainBabelOptions.babelrc = options.babelrc !== false
+  } else {
+    mainBabelOptions.babelrc = false
+  }
+
+  // Add our default preset if the no "babelrc" found.
+  if (!mainBabelOptions.babelrc) {
+    mainBabelOptions.presets.push(require.resolve('../babel/preset'))
+  }
+
+  return mainBabelOptions
+}
+
+function externalsConfig (dir, isServer) {
+  const externals = []
+
+  if (!isServer) {
+    return externals
+  }
+
+  if (fs.existsSync(nextNodeModulesDir)) {
+    externals.push(nodeExternals({
+      modulesDir: nextNodeModulesDir,
+      includeAbsolutePaths: true,
+      whitelist: [/es6-promise|\.(?!(?:js|json)$).{1,5}$/i]
+    }))
+  }
+
+  const dirNodeModules = path.join(dir, 'node_modules')
+  if (fs.existsSync(dirNodeModules)) {
+    nodeExternals({
+      modulesDir: dirNodeModules,
+      includeAbsolutePaths: true,
+      whitelist: [/es6-promise|\.(?!(?:js|json)$).{1,5}$/i]
+    })
+  }
+
+  return externals
+}
+
+export default async function baseConfig (dir, {dev = false, isServer = false, buildId, config}) {
+  const extractCSS = new ExtractTextPlugin({
+    filename: 'static/style.css',
+    disable: dev
+  })
+
+  const cssLoader = {
```

Comment:
Can't we do this all css-loader stuff in the user-land. I mean via the custom webpack config.

We can ship a new package to support basic CSS stuff. So, users can include them in their webpack config or remix as the need.

Here's why?

CSS is a warzone. Everyone has their own config. We need to provide absolute minimum inside Next.js

For an example, we can have  `ExtractTextPlugin` support in core.